src/Resources/config/actions.php

```diff
@@ -28,7 +29,9 @@
     $containerConfigurator->services()
 
         ->set('sonata.admin.action.dashboard', DashboardAction::class)
+            // NEXT_MAJOR: Remove public and sonata.container.private tag.
             ->public()
+            ->tag(AliasDeprecatedPublicServicesCompilerPass::PRIVATE_TAG_NAME, ['version' => '3.x'])
```

Comment:
The deprecation you were receiving when going to the dashboard I think it's because they need to be public because they (controller or actions) are fetched from the container, so please revert the changes from `actions.php`, they are fine as public and they will be in `4.0`.